### PR TITLE
Prevent Tab Switcher from increasing Tab refcount

### LINKED ISSUE #8651: Tabs are not released by TabSwitcher
# Environment

```none
Dev Latest
```

# Steps to reproduce
1. Measure memory
2. Create 10 tabs
3. Open Tab Switcher
4. Close 10 tabs

# Expected behavior
Memory is back to normal

# Actual behavior
Memory remains high until Command Palette (action mode) is open.

Same test without opening tab switcher returns memory usage back to normal.

# Observations
1. Heap analysis shows that `TerminalTab` objects are still alive. Most probably due to `CommandPalette` still holding a reference to them somehow (I believe this semi-leak exists in 1.5 or earlier, but became painful since introduction of `TabPaletteItem`, as now we hold a reference to a tab rather than `SwitchToTabCommand`)
2. We do not not reset tabs in Palette upon tab removal. This is something  we need to fix. But it is insufficient to explain this "leak" - we do call `SetTabs` on the following invocations of TabSwitcher and yet it does not  release the memory
3. I tried to track when the actual memory release happens, and it looks that it is insufficient to switch to the action mode in the code, we really need to make the palette visible before the memory is released. This led me to the thought that the view is somehow holding the ref-count, although the model has already changed.
4. I blamed `x:Bind` not handling `PropertyChanged` ref-counting correctly (like here: https://github.com/microsoft/microsoft-ui-xaml/issues/934). But removing the `INotifyPropertyChanged` completely and even converting the bindings to `OneTime` didn't help. So now I am not sure if this is the right direction

### PR DESCRIPTION
**Pull request by Don-Vito**
Fix `TabPaletteItem` to hold only a weak reference to a tab.
This way we guarantee that the refcount of the closed tab 
gets to 0 immediately
(and that command palette cannot "raise it from the dead").

While this seems a correct thing to do, 
it is still not clear why the `FilteredCommand` itself 
(the one holding the `TabPaletteItem`) doesn't get released
until the UI is refreshed.

There is an impact of not registering to PropertyChanged event:
if the tab title changes during Tab Switcher navigation
the Tab Switcher item won't be updated immediately
(the change will apply next time the Tab Switcher is open).

Due to this change we need to make sure that the tabs binding
in https://github.com/microsoft/terminal/pull/8427
doesn't break the title / icon update.

## Validation Steps Performed
* Manual testing

Closes #8651

### CODE REVIEW

File: src/cascadia/TerminalApp/TabPaletteItem.cpp [hunk 1]
```diff
@@ -19,27 +19,9 @@ using namespace winrt::Microsoft::Terminal::Settings::Model;
...
-                if (changedProperty == L"Title")
-                {
-                    item->Name(senderTab.Title());
-                }
-                else if (changedProperty == L"Icon")
-                {
-                    item->Icon(senderTab.Icon());
-                }
-            }
-        });
```
Comment (DHowett) [inline src/cascadia/TerminalApp/TabPaletteItem.cpp:43]: are we going to be upset that we don't have this?

Comment (Don-Vito) [reply src/cascadia/TerminalApp/TabPaletteItem.cpp:43]: @DHowett - happy new year! Missed you! 😊 

Yes, we will be a upset a bit, as if the switcher is already visible the change in the tab's name / icon won't get reflected immediately. The update will become visible only the next time the tab switcher is open.

We definitely need to find a solution for this, but I assumed this one is urgent enough to restore the functionality in some follow up.

**Review by zadjii-msft — APPROVED** [review]
I'm surprised that we need to remove the `_tab.PropertyChanged` handler too. I'd expect that wouldn't create a strong ref to the Tab - that kinda makes me worried that we have other such leaks elsewhere

**Comment by Don-Vito** [comment]
> 
> 
> I'm surprised that we need to remove the `_tab.PropertyChanged` handler too. I'd expect that wouldn't create a strong ref to the Tab - that kinda makes me worried that we have other such leaks elsewhere

@zadjii-msft - know what.. I only read about this.. but never tried not to remove. I will check and update you.

**Comment by DHowett** [comment]
I love this. Thanks so much. Happy new year indeed!

**Review by DHowett — APPROVED** [review]

**Pull request merged**